modules/default/calendar/calendarfetcherutils.js

```diff
@@ -382,52 +226,52 @@ const CalendarFetcherUtils = {
 								Log.debug("have a recurrence match for dateKey=", dateKey);
 								// We found an override, so for this recurrence, use a potentially different title, start date, and duration.
 								curEvent = curEvent.recurrences[dateKey];
-								curEvent.start = new Date(new Date(curEvent.start.valueOf()).getTime());
-								curEvent.end = new Date(new Date(curEvent.end.valueOf()).getTime());
-								startMoment = CalendarFetcherUtils.getAdjustedStartMoment(curEvent.start, event);
-								endMoment = CalendarFetcherUtils.getAdjustedStartMoment(curEvent.end, event);
-								date = curEvent.start;
-								curDurationMs = new Date(endMoment).valueOf() - startMoment.valueOf();
+								// Some event start/end dates don't have timezones
+								if (curEvent.start.tz) {
+									recurringEventStartMoment = moment(curEvent.start).tz(curEvent.start.tz).tz(CalendarFetcherUtils.getLocalTimezone());
+								} else {
+									recurringEventStartMoment = moment(curEvent.start).tz(CalendarFetcherUtils.getLocalTimezone());
+								}
+								if (curEvent.end.tz) {
+									recurringEventEndMoment = moment(curEvent.end).tz(curEvent.end.tz).tz(CalendarFetcherUtils.getLocalTimezone());
+								} else {
+									recurringEventEndMoment = moment(curEvent.end).tz(CalendarFetcherUtils.getLocalTimezone());
+								}
 							} else {
 								Log.debug("recurrence key ", dateKey, " doesn't match");
 							}
 						}
 						// If there's no recurrence override, check for an exception date.  Exception dates represent exceptions to the rule.
 						if (curEvent.exdate !== undefined) {
-							Log.debug("have datekey=", dateKey, " exdates=", curEvent.exdate);
+							console.log("have datekey=", dateKey, " exdates=", curEvent.exdate);
```

Comment:
should use Log.

Reply:
Oops, well spotted. I will check if I left any other console.log statements or other debug lines in there